internals/feature_helpers.py

```diff
@@ -312,63 +312,149 @@ def get_in_milestone(milestone: int,
+    desktop_origin_trials_ids = list(set(
+        [s.feature_id for s in desktop_origin_trial_future.result()]))
+    android_origin_trials_ids = list(set(
+        [s.feature_id for s in android_origin_trial_future.result()]))
+    webview_origin_trials_ids = list(set(
+        [s.feature_id for s in webview_origin_trial_future.result()]))
+    desktop_dev_trials_ids = list(set(
+        [s.feature_id for s in desktop_dev_trial_future.result()]))
+    android_dev_trials_ids = list(set(
+        [s.feature_id for s in android_dev_trial_future.result()]))
+
+    # Query for FeatureEntry entities that match the stage feature IDs.
+    # Querying with an empty list will raise an error, so check if each
+    # list is not empty first.
+    if desktop_shipping_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in desktop_shipping_ids]))
+      q = q.order(FeatureEntry.name)
+      desktop_shipping_future = q.fetch_async()
+    else:
+      desktop_shipping_future = None
+    if android_only_shipping_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in android_only_shipping_ids]))
+      q = q.order(FeatureEntry.name)
+      android_only_shipping_future = q.fetch_async()
+    else:
+      android_only_shipping_future = None
+    if desktop_origin_trials_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in desktop_origin_trials_ids]))
+      q = q.order(FeatureEntry.name)
+      desktop_origin_trial_future = q.fetch_async()
+    else:
+      desktop_origin_trial_future = None
+    if android_origin_trials_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in android_origin_trials_ids]))
+      q = q.order(FeatureEntry.name)
+      android_origin_trial_future = q.fetch_async()
+    else:
+      android_origin_trial_future = None
+    if webview_origin_trials_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in webview_origin_trials_ids]))
+      q = q.order(FeatureEntry.name)
+      webview_origin_trial_future = q.fetch_async()
+    else:
+      webview_origin_trial_future = None
+    if desktop_dev_trials_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in desktop_dev_trials_ids]))
+      q = q.order(FeatureEntry.name)
+      desktop_dev_trial_future = q.fetch_async()
+    else:
+      desktop_dev_trial_future = None
+    if android_dev_trials_ids:
+      q = FeatureEntry.query(FeatureEntry.key.IN(
+          [ndb.Key('FeatureEntry', id) for id in android_dev_trials_ids]))
+      q = q.order(FeatureEntry.name)
+      android_dev_trial_future = q.fetch_async()
+    else:
+      android_dev_trial_future = None
+
+    desktop_shipping_features = ([] if desktop_shipping_future is None
```

Comment:
Likewise, a tiny function could help here too.  It would return [] or the future result. E.g.,
desktop_shipping_features = _resolve_future(desktop_shipping_future)